lucene/core/src/java/org/apache/lucene/util/fst/Util.java

```diff
@@ -460,11 +460,6 @@ public void addStartPaths(FST.Arc<T> node, T startOutput, boolean allowEmptyStri
           continue;
         }
 
-        if (results.size() == topN-1 && maxQueueDepth == topN) {
-          // Last path -- don't bother w/ queue anymore:
-          queue = null;
```

Comment:
Whoa, was this opto breaking something?  I guess if this final path is filtered out, we still need the queue?  Have you run the suggest benchmarks to see if removing this opto hurt performance?

Reply:
As far as I understand this optimization assumes we surely accept (and collect) the path later in L516s acceptResult(), which always seems to be the case for collectors that don't reject, but if the collector that is eventually called via NRTSuggesters acceptResult() chooses to reject this option, we were losing expected results. This surfaced in the prefix completion tests I added. @jimczi might be able to explain this a bit better than me.

> Have you run the suggest benchmarks to see if removing this opto hurt performance?
No, where are they and how can I run them?